agentos/cli.py

```diff
@@ -236,12 +232,10 @@ def freeze(component_name, registry_file, force, use_venv):
           the same commit
         * There are no uncommitted changes in the local repo
     """
-    venv = VirtualEnv.from_registry_file(registry_file, component_name)
-    venv.set_environment_handling(use_venv)
-    with venv:
-        component = Component.from_registry_file(registry_file, component_name)
-        frozen_reg = component.to_frozen_registry(force=force)
-        print(yaml.dump(frozen_reg.to_dict()))
+    component = Component.from_registry_file(registry_file, component_name)
+    component.set_environment_handling(use_venv)
```

Comment:
Hmm, yeah, that seems better.  

One (related) awkwardness is that every time we add a new argument to `Component.__init__()`, we (usually) need to add that argument to all the `Component.from_*()` methods, even if it's just passed through to the Component constructor.

I'm thinking there might be a way to streamline this when we get to the point of making the Component constructor the one source of truth for the spec structure.

But for now, it does seem nicer to add an optional argument to the `Component.from_*()` methods instead of a separate setter/getter.  Will fix!